crates/ironrdp-web/src/clipboard/transaction.rs

```diff
@@ -1,62 +1,61 @@
-use wasm_bindgen::prelude::*;
+use iron_remote_desktop::{ClipboardContent, ClipboardTransaction};
+use serde::Serialize;
+use wasm_bindgen::JsValue;
 
 /// Object which represents complete clipboard transaction with multiple MIME types.
-#[wasm_bindgen]
-#[derive(Debug, Default, Clone)]
-pub struct ClipboardTransaction {
-    contents: Vec<ClipboardContent>,
+#[derive(Debug, Default, Clone, Serialize)]
+pub(crate) struct RdpClipboardTransaction {
+    contents: Vec<RdpClipboardContent>,
 }
 
-impl ClipboardTransaction {
-    pub fn contents(&self) -> &[ClipboardContent] {
+impl RdpClipboardTransaction {
+    pub(crate) fn contents(&self) -> &[RdpClipboardContent] {
         &self.contents
     }
 
-    pub fn clear(&mut self) {
+    pub(crate) fn clear(&mut self) {
         self.contents.clear();
     }
 }
 
-#[wasm_bindgen]
-impl ClipboardTransaction {
-    pub fn init() -> Self {
+impl ClipboardTransaction for RdpClipboardTransaction {
+    type ClipboardContent = RdpClipboardContent;
+
+    fn init() -> Self {
         Self { contents: Vec::new() }
     }
 
-    pub fn add_content(&mut self, content: ClipboardContent) {
+    fn add_content(&mut self, content: Self::ClipboardContent) {
         self.contents.push(content);
     }
 
-    pub fn is_empty(&self) -> bool {
+    fn is_empty(&self) -> bool {
         self.contents.is_empty()
     }
 
-    #[wasm_bindgen(js_name = content)]
-    pub fn js_contents(&self) -> js_sys::Array {
-        js_sys::Array::from_iter(
-            self.contents
-                .iter()
-                .map(|content: &ClipboardContent| JsValue::from(content.clone())),
-        )
+    fn js_contents(&self) -> js_sys::Array {
+        js_sys::Array::from_iter(self.contents.iter().map(|content: &RdpClipboardContent| {
+            serde_wasm_bindgen::to_value(&content).expect("Failed to convert clipboard transaction value into JsValue")
+        }))
     }
 }
 
-impl FromIterator<ClipboardContent> for ClipboardTransaction {
-    fn from_iter<T: IntoIterator<Item = ClipboardContent>>(iter: T) -> Self {
+impl FromIterator<RdpClipboardContent> for RdpClipboardTransaction {
+    fn from_iter<T: IntoIterator<Item = RdpClipboardContent>>(iter: T) -> Self {
         Self {
             contents: iter.into_iter().collect(),
         }
     }
 }
 
-#[derive(Debug, Clone)]
-pub enum ClipboardContentValue {
+#[derive(Debug, Clone, Serialize)]
```

Comment:
question: What is the benefit of using `Serialize` over the pre-existing `js_value` function?

Reply:
Removed usage of `serde`. It is required to add `#[wasm_bindgen]` on the struct so it can be made into `JsValue` without serde.

Reply:
Improved in https://github.com/Devolutions/IronRDP/pull/760